apps/web/pages/api/auth/csrf.ts

```diff
@@ -0,0 +1,25 @@
+import { serialize } from "cookie";
+import { randomBytes } from "crypto";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  console.log("✨ Getting CSRF token...");
+  if (req.method === "GET") {
+    const token = randomBytes(28).toString("hex");
+    res.setHeader(
+      "Set-Cookie",
+      serialize("csrf_token", token, {
+        httpOnly: false, // important for reading cookie on the client
+        maxAge: undefined, // expire with session
+        sameSite: "strict",
```

Comment:
Note that in embed we are in third party context and this cookie would not be sent from within the embed iframe potentially breaking embeds

Reply:
This is a good point!